libs/prelude/Prelude/Functor.idr

```diff
@@ -6,15 +6,15 @@ import Prelude.Basics
 %access public export
 
 ||| Functors allow a uniform action over a parameterised type.
-||| @ f a parameterised type 
+||| @ f a parameterised type
 interface Functor (f : Type -> Type) where
-    ||| Apply a function across everything of type 'a' in a 
+    ||| Apply a function across everything of type 'a' in a
     ||| parameterised type
     ||| @ f the parameterised type
     ||| @ func the function to apply
     map : (func : a -> b) -> f a -> f b
 
-infixl 4 <$>
+infixr 4 <$>
```

Comment:
Since `(->) a` is not a functor in Idris, chaining doesn't work with the current fixity.
```idris example
-- this now works:
(+4) <$> (*2) <$> Just (the Int 10)
```
An alternate way to solve this would be to define another version of `(<$>)`, as an alias for `(.)`.